src/statistics/data.jl

```diff
@@ -66,3 +72,32 @@ function median_heuristic(d)
 
   min(m, l)
 end
+
+function mode_heuristic(d)
+
+  D = dist_matrix_random_sample(d)
+  n = size(D, 1)
+  D_upper_sorted = sort(collect(D[i,j] for i in 1:n for j in 1:n if i > j))
+
+  x_n = D_upper_sorted
+
+  while length(x_n) >= 4
+    i = length(x_n) ÷ 2
+    x_n_1 = x_n[1:i]
+    x_n_2 = x_n[i + 1:end]
+    x_n = [x_n_1, x_n_2][argmin([(x_n_1[end] - x_n_1[1]), (x_n_2[end] - x_n_2[1])])]
```

Comment:
This line could be refactored for increased readability. There are two steps being performed: search and selection. Also, when you use `end` for the last index you can also use `begin` for the first for consistency.

Reply:
Alternatively, you can use `last(x) - first(x)`, it is a matter of taste. In some cases using `begin/end` indices makes things clearer to read and in some other cases `last/first` functions are preferred.